lxc/Deploy.sh

```diff
@@ -0,0 +1,188 @@
+#   description: ""
+#   devices:
+#     root:
+#       path: /
+#       pool: default
+#       type: disk
+#   name: default
+# projects: []
+# cluster: null
+
+# Possible errors:
+
+# 1) If you get keyring error like: The specified keyring /var/cache/lxc/debian/archive-key.gpg may be incorrect or out of date
+# Just download the new keys and import it with:
+#
+# wget "https://ftp-master.debian.org/keys/release-10.asc"
+# sudo gpg --no-default-keyring --keyring /var/cache/lxc/debian/archive-key.gpg --import release-10.asc
+
+# 2) If you get an error tryint to run virsh commands like:
+# error: failed to connect to the hypervisor
+# error: Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory
+#
+# Install kvm with:
+# sudo apt install qemu qemu-kvm libvirt-clients libvirt-daemon-system virtinst bridge-utils
+# sudo systemctl enable libvirtd
+# sudo systemctl start libvirtd
+
+#export USER=abelyak
+
+export CONTAINERS=10
+export NETPREF=192.168.122
+
+
+if [[ ! $USER ]]; then
+   echo "Specify USER variable"
+   exit 1
+fi
+
+if [ `whoami` != 'root' ]; then
+    echo "Run as root"
+    exit 2
+fi
+
+
+
+function setup {
+	echo "01. Create"
+	for (( i=1; i<=$CONTAINERS; i++ )) do lxc-create -n n$i -t debian -- --release buster; done
+
+	echo "02. Config network"
+	for (( i=1; i<=$CONTAINERS; i++ )) do
+
+		sed -i -E 's/(.+lxc\.net.+)/#\1/' /var/lib/lxc/n${i}/config
+
+		cat >>/var/lib/lxc/n${i}/config <<EOF
+
+# Network config
+lxc.net.0.type = veth
+lxc.net.0.flags = up
+lxc.net.0.link = virbr0
+lxc.net.0.hwaddr = 00:1E:62:AA:AA:$(printf "%02x" $i)
+EOF
+	done
+
+	echo "03. Network bindings"
+	for (( i=1; i<=$CONTAINERS; i++ )) do
+	  virsh net-update --current default add-last ip-dhcp-host "<host mac=\"00:1E:62:AA:AA:$(printf "%02x" $i)\" name=\"n${i}\" ip=\"$NETPREF.1$(printf "%02d" $i)\"/>"
+	done
+
+	echo "04. Start network"
+	virsh net-autostart default;
+	virsh net-start default
+
+	echo "05. Add hostnames"
+	#echo -e "nameserver 192.168.122.1\n$(cat /etc/resolv.conf)" > /etc/resolv.conf
+
+	if [ ! grep -q "Jepsen hosts" /etc/hosts ]; then
+		echo "Jepsen hosts" >> /etc/hosts
+
+		for (( i=1; i<=$CONTAINERS; i++ )) do
+			echo -e "${NETPREF}.${i} n${1}" >> /etc/hosts
+		done
+    fi
+
+	# NOTE: systemd не даёт записывать данные в `/etc/resolv.conv`, они стираются при перезагрузке
+	# пробую тупо добавить адреса в /etc/hosts?
+
+	echo "06. Setup dhcp client dns"
+	echo -e "prepend domain-name-servers ${NETPREF}.1;" >>/etc/dhcp/dhclient.conf
+	service networking restart
+
+	echo "07. Copy keys to the lxc containers"
+	export KEYFILE=/home/$USER/.ssh/id_rsa.pub
+
+	for (( i=1; i<=$CONTAINERS; i++ )) do
+	  mkdir -p /var/lib/lxc/n${i}/rootfs/root/.ssh
+	  chmod 700 /var/lib/lxc/n${i}/rootfs/root/.ssh/
+	  cp $KEYFILE /var/lib/lxc/n${i}/rootfs/root/.ssh/authorized_keys
+	  chmod 644 /var/lib/lxc/n${i}/rootfs/root/.ssh/authorized_keys
+	done
+
+	start
+
+	echo "08. Setup SSHD in lxc"
+	#!/bin/bash
```

Comment:
Copy-paste? No need to put shebang in the middle of the script. :)

Reply:
Removed